backstory/src/main/java/com/google/sps/images/VisionImagesManager.java

```diff
@@ -55,52 +57,56 @@ public VisionImagesManager(ImageAnnotatorClient imageAnnotatorClient) {
 
-    // TODO: parallelize calls to detectLabelsFromImageBytes() since it requires a network call.
-    for (byte[] rawImageData : imagesAsByteArrays) {
-      List<EntityAnnotation> labelAnnotations = detectLabelsFromImageBytes(rawImageData);
-      AnnotatedImage annotatedImage = new AnnotatedImage(rawImageData, labelAnnotations);
-      annotatedImages.add(annotatedImage);
-    }
+    // add the features we want (labels & landmarks to list)
+    List<Feature> features = new ArrayList<Feature>();
 
-    return annotatedImages;
-  }
+    Feature labelFeature = Feature.newBuilder().setType(Feature.Type.LABEL_DETECTION).build();
+    Feature landmarkFeature = Feature.newBuilder().setType(Feature.Type.LANDMARK_DETECTION).build();
 
-  /**
-   * Creates and returns label annotations for the image represented in bytes, using the Vision API.
-   *
-   * @param bytes The raw image byte data for the image from which labels will be annotated.
-   * @return The list of label annotations related to the image, with each individual label being
-   *     represented as an EntityAnnotation object.
-   */
-  private List<EntityAnnotation> detectLabelsFromImageBytes(byte[] bytes) throws IOException {
-    List<AnnotateImageRequest> requests = new ArrayList<>();
-    List<EntityAnnotation> labels;
+    features.add(labelFeature);
+    features.add(landmarkFeature);
 
-    Image img = Image.newBuilder().setContent(ByteString.copyFrom(bytes)).build();
-    Feature feat = Feature.newBuilder().setType(Feature.Type.LABEL_DETECTION).build();
-    AnnotateImageRequest request =
-        AnnotateImageRequest.newBuilder().addFeatures(feat).setImage(img).build();
-    requests.add(request);
+    // iterate through the images to build the request
+    int size = imagesAsByteArrays.size();
+
+    for (int i = 0; i < size; i++) {
+      Image image =
+          Image.newBuilder().setContent(ByteString.copyFrom(imagesAsByteArrays.get(i))).build();
+      AnnotateImageRequest request =
+          AnnotateImageRequest.newBuilder().addAllFeatures(features).setImage(image).build();
+      requests.add(request);
+    }
+
+    // TODO: parallelize calls to detectLabelsFromImageBytes() since it requires a network call.
 
     // The invocation of batchAnnotateImages() makes a network call.
-    BatchAnnotateImagesResponse response = imageAnnotatorClient.batchAnnotateImages(requests);
-    List<AnnotateImageResponse> responses = response.getResponsesList();
+    BatchAnnotateImagesResponse batchResponse = imageAnnotatorClient.batchAnnotateImages(requests);
+    List<AnnotateImageResponse> responses = batchResponse.getResponsesList();
 
     // TODO: parallelize calls to detectLabelsFromImageBytes() since it requires a network call.
-    if (!(responses.size() == 1)) {
-      throw new IllegalArgumentException(
-          "detectLabelsFromImageBytes only supports analytics on one image");
+    if (!(responses.size() == size)) {
```

Comment:
How about renaming `size` to `imageCount` or `numberOfImages`.

I had to go back up to figure out what `size` is.

Reply:
I no longer use it multiple times (used it above in the for loop), so I'm just going to call `imagesAsByteArrays.size()` here instead. I hope this helps.

Reply:
Done